File: NavHub/src/api/siteConfig.ts
import request from './request';
import type { SiteConfig } from '@/hooks/useSiteConfig';

const USE_MOCK = false;

// Mock数据
const mockSiteConfig: SiteConfig = {
  siteName: 'NavHub',
  siteTitle: 'NavHub - 精选网站导航',
  siteLogo: '/assets/logo.png',
  siteDescription: '发现优质网站，提升工作效率。汇聚设计、开发、工具等各类精选网站资源。',
  icpRecord: '',
  publicSecurityRecord: '',
  publicSecurityRecordUrl: '',
};

/**
 * 站点配置API
 */
export const siteConfigApi = {
  /**
   * 获取站点配置
   */
  async get(): Promise<SiteConfig> {
    if (USE_MOCK) {
      return mockSiteConfig;
    }
    const result = await request.get<SiteConfig>('/site-config');
    return result;
  },

  /**
   * 更新站点配置（管理员）
   */
  async update(config: Partial<SiteConfig>): Promise<SiteConfig> {
    if (USE_MOCK) {
      return { ...mockSiteConfig, ...config };
    }
    
    // 确保 siteLogo 字段被包含在请求中
    // 如果 siteLogo 是 undefined，不包含该字段；如果是 null 或空字符串，也要传递
    const requestData: any = {
      ...config,
    };
    
    // 明确设置 siteLogo 字段，确保它被包含在请求中
    if (config.siteLogo !== undefined) {
      requestData.siteLogo = config.siteLogo;
    }
    
    const result = await request.put<SiteConfig>('/site-config', requestData);
    return result;
  },
};

File: NavHub-Con/src/main/java/com/navhub/entity/SiteConfig.java
package com.navhub.entity;

import lombok.Data;
import java.time.LocalDateTime;

/**
 * 站点配置实体
 */
@Data
public class SiteConfig {
    /**
     * 配置ID（固定值：site_config_001）
     */
    private String id;
    
    /**
     * 站点名称
     */
    private String siteName;
    
    /**
     * 站点标题
     */
    private String siteTitle;
    
    /**
     * 站点Logo（URL或base64）
     */
    private String siteLogo;
    
    /**
     * 站点描述
     */
    private String siteDescription;
    
    /**
     * ICP备案号
     */
    private String icpRecord;
    
    /**
     * 公安备案号
     */
    private String publicSecurityRecord;
    
    /**
     * 公安备案链接
     */
    private String publicSecurityRecordUrl;
    
    /**
     * 创建时间
     */
    private LocalDateTime createdAt;
    
    /**
     * 更新时间
     */
    private LocalDateTime updatedAt;
}

